lib/kitchen/driver/ec2.rb

```diff
@@ -382,9 +425,57 @@ def hostname(server, interface_type = nil)
       end
 
       def create_ec2_json(state)
-        instance.transport.connection(state).execute(
-          "sudo mkdir -p /etc/chef/ohai/hints;sudo touch /etc/chef/ohai/hints/ec2.json"
-        )
+        if windows_os?
+          cmd = "mkdir c:\\chef\\ohai\\hints; echo $null >> c:\\chef\\ohai\\hints\\ec2.json"
+        else
+          cmd = "sudo mkdir -p /etc/chef/ohai/hints;sudo touch /etc/chef/ohai/hints/ec2.json"
+        end
+        instance.transport.connection(state).execute(cmd)
+      end
+
+      def default_windows_user_data
+        #Preparing custom static admin user if we defined something other than Administrator
+        customAdminScript = ''
+        if !(instance.transport[:username] =~ /administrator/i) && instance.transport[:password]
+          customAdminScript = Kitchen::Util.outdent!(<<-EOH)
+          "Disabling Complex Passwords" >> $logfile
+          $seccfg = [IO.Path]::GetTempFileName()
+          & secedit.exe /export /cfg $seccfg >> $logfile
+          (Get-Content $seccfg) | Foreach-Object {$_ -replace "PasswordComplexity\\s*=\\s*1", "PasswordComplexity = 0"} | Set-Content $seccfg
+          & secedit.exe /configure /db $env:windir\\security\\new.sdb /cfg $seccfg /areas SECURITYPOLICY >> $logfile
+          & cp $seccfg "c:\\"
+          & del $seccfg
+          $username="#{instance.transport[:username]}"
+          $password="#{instance.transport[:password]}"
+          "Creating static user: $username" >> $logfile
+          & net.exe user /y /add $username $password >> $logfile
+          "Adding $username to Administrators" >> $logfile
+          & net.exe localgroup Administrators /add $username >> $logfile
+          EOH
+        end
+
+        # Returning the fully constructed PowerShell script to user_data
+        Kitchen::Util.outdent!(<<-EOH)
+        <powershell>
+        $logfile="C:\\Program Files\\Amazon\\Ec2ConfigService\\Logs\\kitchen-ec2.log"
+        #PS Remoting and & winrm.cmd basic config
+        Enable-PSRemoting -Force -SkipNetworkProfileCheck
+        & winrm.cmd quickconfig -q >> $logfile
+        & winrm.cmd set winrm/config '@{MaxTimeoutms="1800000"}' >> $logfile
+        & winrm.cmd set winrm/config/winrs '@{MaxMemoryPerShellMB="512"}' >> $logfile
+        & winrm.cmd set winrm/config/winrs '@{MaxShellsPerUser="50"}' >> $logfile
+        #Client settings
+        & winrm.cmd set winrm/config/client/auth '@{Basic="true"}' >> $logfile
+        #Server settings
+        & winrm.cmd set winrm/config/service/auth '@{Basic="true"}' >> $logfile
+        & winrm.cmd set winrm/config/service '@{AllowUnencrypted="true"}' >> $logfile
+        & winrm.cmd set winrm/config/winrs '@{MaxMemoryPerShellMB="1024"}' >> $logfile
+        #Firewall Config
+        & netsh.exe advfirewall set publicprofile state off  >> $logfile
```

Comment:
I'm not saying it is an advantage, just in one line we turn off the firewall and then we update a rule that doesn't matter because we turned the firewall off.

We should just need the below rule and not turn off the firewall with the first rule.